rm/rm-server/src/test/java/functionaltests/housekeeping/NodesHouseKeepingPropertiesTest.java

```diff
@@ -0,0 +1,95 @@
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package functionaltests.housekeeping;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.io.File;
+
+import org.junit.Test;
+import org.ow2.proactive.resourcemanager.common.event.RMEventType;
+import org.ow2.proactive.resourcemanager.core.properties.PAResourceManagerProperties;
+import org.ow2.proactive.resourcemanager.frontend.ResourceManager;
+
+import functionaltests.nodesrecovery.NodesRecoveryProcessHelper;
+import functionaltests.nodesrecovery.RecoverInfrastructureTestHelper;
+import functionaltests.utils.RMFunctionalTest;
+import functionaltests.utils.RMTHelper;
+
+
+public class NodesHouseKeepingPropertiesTest extends RMFunctionalTest {
+
+    private static final String RM_CONFIG_FILE_PATH = "/functionaltests/config/functionalTRMProperties-remove-nodes-unavailable-for-five-min-every-min.ini";
+
+    private static final String NODE_SOURCE_NAME = "LocalNodeSource-" +
+                                                   NodesHouseKeepingPropertiesTest.class.getSimpleName();
+
+    private static final int NODE_NUMBER = 4;
+
+    private static final String EVERY_MINUTE_CRON_EXPRESSION = "* * * * *";
+
+    private static final int REMOVE_DOWN_NODES_AFTER_MINUTES = 2;
+
+    @Test
+    public void testDownNodesAreRemovedAfterUnavailableMaxPeriodExpired() throws Exception {
+        String rmConfigurationFile = new File(RMTHelper.class.getResource(RM_CONFIG_FILE_PATH)
+                                                             .toURI()).getAbsolutePath();
+        this.rmHelper.startRM(rmConfigurationFile);
+        ResourceManager resourceManager = this.rmHelper.getResourceManager();
+
+        assertThat(PAResourceManagerProperties.RM_UNAVAILABLE_NODES_REMOVAL_FREQUENCY.getValueAsString()).isEqualTo(EVERY_MINUTE_CRON_EXPRESSION);
+        assertThat(PAResourceManagerProperties.RM_UNAVAILABLE_NODES_MAX_PERIOD.getValueAsInt()).isEqualTo(REMOVE_DOWN_NODES_AFTER_MINUTES);
+
+        assertThat(this.rmHelper.isRMStarted()).isTrue();
+        assertThat(resourceManager.getState().getAllNodes().size()).isEqualTo(0);
+
+        this.rmHelper.createNodeSource(NODE_SOURCE_NAME, NODE_NUMBER);
+        assertThat(resourceManager.getState().getAllNodes().size()).isEqualTo(4);
+
+        findNodeProcessAndKill();
+
+        this.rmHelper.waitForAnyMultipleNodeEvent(RMEventType.NODE_STATE_CHANGED, NODE_NUMBER);
+        assertThat(resourceManager.listAliveNodeUrls()).isEmpty();
+        assertThat(resourceManager.getState().getAllNodes().size()).isEqualTo(4);
+
+        int expiredDownNodeMillis = REMOVE_DOWN_NODES_AFTER_MINUTES * 60 * 1000;
+        int nextDownNodesRemovalAttemptMillis = 60 * 1000;
+        Thread.sleep(expiredDownNodeMillis + nextDownNodesRemovalAttemptMillis);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)